ui/app/src/components/SiteConfigurationManagement/SiteConfigurationManagement.tsx

```diff
@@ -156,6 +141,35 @@ export function SiteConfigurationManagement(props: SiteConfigurationManagementPr
     });
   });
 
+  useUnmount(() => {
+    if (!refs.current.disabledSaveButton && mountMode === 'page') {
+      sessionStorage.setItem(
+        sessionStorageKey,
+        JSON.stringify({
+          content: editorRef.current.getValue(),
+          selectedConfigFile: refs.current.selectedConfigFile
+        })
+      );
+      const eventId = 'unsavedConfigurationChangesConfirmation';
+      dispatch(
+        showConfirmDialog({
+          body: <FormattedMessage defaultMessage="You left unsaved changes. Go back and continue editing?" />,
+          onCancel: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: eventId, button: 'cancel' })]),
+          onOk: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: eventId, button: 'ok' })]),
+          okButtonText: <FormattedMessage defaultMessage="Continue editing" />,
+          cancelButtonText: <FormattedMessage defaultMessage="Discard changes" />
+        })
+      );
+      createCustomDocumentEventListener<{ button: 'ok' | 'cancel' }>(eventId, ({ button }) => {
+        if (button === 'ok') {
+          navigate('/configuration');
```

Comment:
```suggestion
          navigate(Routes.Configuration);
```

Reply:
We may need a different way of retrieving this configuration url. Will ping you tomorrow